ports/nrf/common-hal/microcontroller/Pin.c

```diff
@@ -24,16 +24,134 @@
  * THE SOFTWARE.
  */
 
-#include "common-hal/microcontroller/Pin.h"
+#include "shared-bindings/microcontroller/Pin.h"
+
 #include "nrf_gpio.h"
 #include "py/mphal.h"
 
-bool common_hal_mcu_pin_is_free(const mcu_pin_obj_t *pin) {
-    return true;
-}
+#include "nrf/pins.h"
+#include "supervisor/shared/rgb_led_status.h"
+
+#ifdef MICROPY_HW_NEOPIXEL
+bool neopixel_in_use;
+#endif
+#ifdef MICROPY_HW_APA102_MOSI
+bool apa102_sck_in_use;
+bool apa102_mosi_in_use;
+#endif
+#ifdef SPEAKER_ENABLE_PIN
+bool speaker_enable_in_use;
+#endif
+
+// Bit mask of claimed pins on each of up to two ports. nrf52832 has one port; nrf52840 has two.
+STATIC uint32_t claimed_pins[2];
```

Comment:
You can use `GPIO_COUNT` here to be more future-proof.

Reply:
Thanks. For some reason I thought `GPIO_COUNT` wasn't defined for nRF52832, but it is. I'll do that.